### ISSUE TITLE
MQTT_Connect always returns MQTTRecvFailed

### ISSUE THREAD

**Issue by jcracine-cimeq** [aws/aws-iot-device-sdk-embedded-C#1607]
Hi,
Currently using the SDK in a C++ application and I've taken examples from the `mqtt_demo_basic_tls` demos. So I coded a basic C++ wrapper from that. Everything was fine until I tested the communication to the AWS IoT Core. I've setted up the certificates and everything, but I always receive the same error: MQTTRecvFailed. I've upped the logging level to Debug, but this is all I get.

> [AwsMqttCommunication] Establishing a TLS session to \<aws endpoint\>
> [ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:760] Failed to receive data over network: SSL_read failed: ErrorStatus=EVP lib.
> [AwsMqttCommunication] Connection with MQTT broker failed: MQTTRecvFailed

I've searched through the issues and I've found this issue [here](https://github.com/aws/aws-iot-device-sdk-embedded-C/issues/1537) and read everything, but I don't think its a thread issue since I run everything in the same thread.
Also, I tried to use `mosquitto_pub` to see if I was able to communicate and if the problem was my code only, but that didn't work either. 
So I figured it could be because I worked and tested it on a Ubuntu VM, and I tried the moquitto command from a Raspberry Pi I had lying around with the same certificates but I had the same results. Here is the console outputs: 

>  Client <thing name> sending CONNECT
> Error: The connection was lost.

I'm kinda running out of options now so I hoped you could help me again.
Thank you.

edit: Here is the SDK version I'm using and the OS of my VM and the Raspberry Pi.

- SDK Version: 202103.00
- VM OS: Ubuntu 20.04 LTS
- RPi OS: Raspbian GNU/Linux 10 (Buster)

**Comment by muneebahmed10** [comment]
Hi @jcracine-cimeq,

The MQTT Basic TLS demo is not designed to be used with AWS IoT Core as it uses QoS 2 publishes and does not use mutual authentication. Please use the [MQTT Mutual Auth demo](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/main/demos/mqtt/mqtt_demo_mutual_auth) to connect to AWS IoT Core.

Additionally, from your log output the connection is disconnected after sending the CONNECT packet. If this is still the case after running the mutual auth demo, it could be due to misconfigured permissions with AWS IoT. In that case, please ensure that you've followed the steps [here](https://docs.aws.amazon.com/iot/latest/developerguide/iot-gs.html) to configure your account and AWS IoT Thing and your Thing has attached policies with sufficient permissions to connect to AWS IoT Core. The mutual auth demo contains a minimal [sample template policy](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/main/demos/mqtt/mqtt_demo_mutual_auth/aws_iot_policy_example_mqtt.json) that can be used to run the demo with AWS IoT (you'll need to replace your region and account ID).

Let us know if you have any further questions.

**Comment by jcracine-cimeq** [comment]
Ok I've read the `mqtt_demo_mutual_auth` file and the main differences between it and the previous one was the use of certificates when connecting to the broker and using the metric string as the username with MQTT_Connect(). I already placed the certificates paths correctly, but I wasn't using the metric string. Even after adding it, I still had the same errors. Here it just to be sure I wrote it correctly : `?SDK=Ubuntu&Version=20.04 LTS&Platform=PC&MQTTLib=core-mqtt@v1.1.1`.
